### PR TITLE
PT-731 | Indicate if occurrence is full, also support new SeatType

### PR DESCRIPTION
**Pull request by Sartonon**
## Description :sparkles:

- Inidicate if occurrence is full (yello background and tooltip)
- Support new SeatType

### Closes :no_good_woman:
**[PT-731](https://helsinkisolutionoffice.atlassian.net/browse/PT-731):** 

## Screenshots :camera_flash:
Screenshot 2021-01-28 at 16 54 27
Screenshot 2021-01-28 at 13 39 41

## Additional notes :spiral_notepad:

### CODE REVIEW

File: src/domain/occurrences/enrolmentsBadge/__tests__/EnrolmentsBadge.test.tsx [hunk 1]
```diff
@@ -5,7 +5,11 @@ import EnrolmentBadge from '../EnrolmentsBadge';
 
 it('matches snapshot', () => {
   const { container } = render(
-    <EnrolmentBadge acceptedSeatsCount={10} pendingSeatsCount={10} />
+    <EnrolmentBadge
+      acceptedSeatsCount={10}
+      pendingSeatsCount={10}
+      remainingSeatsCount={30}
```
[the remark this answers is not in the document]
Comment (nenonja) [reply src/domain/occurrences/enrolmentsBadge/__tests__/EnrolmentsBadge.test.tsx:11]: Ok they seem to be on EventSummaryPage test perhaps it is ok because it is more end-to-end and this does not have corner case tests.

File: src/domain/event/__tests__/EventSummaryPage.test.tsx [hunk 2]
```diff
@@ -268,3 +278,32 @@ it('shows upcoming and past occurrences', async () => {
     screen.getByRole('heading', { name: 'Menneet tapahtuma-ajat 2 kpl' })
   ).toBeInTheDocument();
```
Comment (nenonja) [inline src/domain/event/__tests__/EventSummaryPage.test.tsx:279]: perhaps here could be added that for event2 that is not full, the `enrolmentFull` class and text are not in the document

File: src/domain/occurrences/occurrencesTable/OccurrencesTable.tsx [hunk 3]
```diff
@@ -148,6 +156,7 @@ const OccurrencesTable: React.FC<Props> = ({
             <EnrolmentsBadge
               acceptedSeatsCount={row.seatsApproved}
```
Comment (nenonja) [inline src/domain/occurrences/occurrencesTable/OccurrencesTable.tsx:157]: There seems to be synonyms: `acceptedSeats === approvedSeats` Which lead to confusion. Synonyms should be avoided in codebase. I'm not sure though is it too big task to rename one to another in this pr? But it would make code cleaner at least to me :)

File: src/domain/app/i18n/fi.json [hunk 4]
```diff
@@ -521,6 +521,8 @@
     "pageTitle": "Tapahtuma-aika",
     "textDateAndTime": "{{date}} klo {{time}}",
     "textAmountOfSeats": "{{count}} paikka",
+    "textAmountOfGroups": "{{count}} ryhmä",
+    "textAmountOfGroups_plural": "{{count}} ryhmää",
```
Comment (nenonja) [inline src/domain/app/i18n/fi.json:525]: I did not find that `textAmountOfGroups_plural` is used anywhere?

File: src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
 import styles from './enrolmentsBadge.module.scss';
 
 interface EnrolmentsBadgeProps {
   acceptedSeatsCount?: number | null;
   pendingSeatsCount?: number | null;
+  remainingSeatsCount?: number | null;
 }
 
 const EnrolmentsBadge: React.FC<EnrolmentsBadgeProps> = ({
   acceptedSeatsCount = 0,
```
[the remark this answers is not in the document]
Comment (nenonja) [reply src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx:15]: Okay I answer to first question to myself: pending === seatsTaken - seatsApproved. Second question is still open to me

**Review by nenonja — COMMENTED** [review]
Almost there some comments! 
Also I was not able to test it as I dont have occurence that would be full...
One thing I noted that voiceover does not read very well the rows here:
Screenshot 2021-02-02 at 10 37 58 AM
It reads first line `[...] nolla nolla valitse" which does not tell much about how mean seats are approved and pending

File: src/domain/app/i18n/fi.json (hunk 4 above)
Comment (Sartonon) [reply src/domain/app/i18n/fi.json:525]: _plural works automatically when count is  0 or more than 1

see: https://www.i18next.com/translation-function/plurals

File: src/domain/app/i18n/en.json [hunk 6]
```diff
@@ -521,6 +521,8 @@
     "textDateAndTime": "{{date}} at {{time}}",
     "textAmountOfSeats": "{{count}} seat",
     "textAmountOfSeats_plural": "{{count}} seats",
+    "textAmountOfGroups": "{{count}} grupp",
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/app/i18n/en.json:525]: good catch :)

File: src/domain/app/i18n/sv.json [hunk 7]
```diff
@@ -567,7 +567,8 @@
       "columnLocation": "Evenemang plats",
       "columnTime": "Tid",
       "labelChooseAllOccurrences": "Välj alla förekomster",
-      "labelChooseOccurrence": "Välj förekomst {{info}}"
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/app/i18n/sv.json:570]: fixed

File: src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx]: Changed

[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx]: good question, can't remember can accepted + pending + remaining be more than seatsCount

File: src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx [hunk 8]
```diff
@@ -1,22 +1,41 @@
...
   pendingSeatsCount = 0,
+  remainingSeatsCount = 0,
 }) => {
+  const { t } = useTranslation();
+  const isOccurrenceFull = remainingSeatsCount === 0;
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx:20]: Changed

File: src/domain/occurrences/enrolmentsBadge/__tests__/EnrolmentsBadge.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/occurrences/enrolmentsBadge/__tests__/EnrolmentsBadge.test.tsx:11]: yeah

File: src/domain/event/__tests__/EventSummaryPage.test.tsx [hunk 9]
```diff
@@ -268,3 +278,32 @@ it('shows upcoming and past occurrences', async () => {
...
+    routes: [`/events/${eventId1}/summary`],
+  });
+
+  await waitFor(() => {
+    expect(screen.queryByText(organisationName)).toBeInTheDocument();
+    expect(screen.queryAllByText(eventName)).toHaveLength(2);
+  });
+
+  const fullOccurrence = screen.queryByRole('row', {
+    name: fullOccurrecneRowText,
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/event/__tests__/EventSummaryPage.test.tsx]: fixed

File: src/domain/event/__tests__/EventSummaryPage.test.tsx [hunk 10]
```diff
@@ -268,3 +278,32 @@ it('shows upcoming and past occurrences', async () => {
...
+  });
+
+  expect(fullOccurrence).toBeInTheDocument();
+
+  const withinFullOccurrence = within(fullOccurrence);
+  const acceptedEnrolments = withinFullOccurrence.getByText('10');
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/event/__tests__/EventSummaryPage.test.tsx]: Updated

File: src/domain/event/__tests__/EventSummaryPage.test.tsx (hunk 2 above)
Comment (Sartonon) [reply src/domain/event/__tests__/EventSummaryPage.test.tsx:288]: improved tests

File: src/domain/occurrences/occurrencesTable/OccurrencesTable.tsx (hunk 3 above)
Comment (Sartonon) [reply src/domain/occurrences/occurrencesTable/OccurrencesTable.tsx]: renamed to approvedSeats

File: src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (nenonja) [reply src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx:20]: I dont find it?

File: src/domain/app/i18n/fi.json (hunk 4 above)
Comment (nenonja) [reply src/domain/app/i18n/fi.json:526]: Nice one, I did not know

File: src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx:20]: strange, I don't know what happened

[the remark this answers is not in the document]
Comment (nenonja) [reply src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx:20]: I couldnt find this change?....

[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/occurrences/enrolmentsBadge/EnrolmentsBadge.tsx]: ok seems like I forget to push latest commit

**Review by nenonja — APPROVED** [review]
Approved, but I was not able to test it as I did not know any occurence that is full. One test is failing because it's broken in master. I had a fix in my pr: https://github.com/City-of-Helsinki/palvelutarjotin-admin/pull/143

**Comment by Sartonon** [comment]
Lets see if it causes problems in the future
